bin/update-extension-files.js

```diff
@@ -0,0 +1,340 @@
+		} )();
+	}
+
+	/**
+	 * Fetch AMP compatible plugins and themes.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async fetchData() {
+		let totalPage = 1;
+		const pluginTerm = 552;
+		const themeTerm = 245;
+		const url = 'https://amp-wp.org/wp-json/wp/v2/ecosystem';
+
+		const queryParams = {
+			ecosystem_types: [ themeTerm, pluginTerm ],
+			per_page: 100,
+			page: 1,
+		};
+
+		do {
+			// eslint-disable-next-line no-await-in-loop
+			const response = await axios.get( url, { params: queryParams } );
+			const items = response.data;
+			totalPage = parseInt( response.headers[ 'x-wp-totalpages' ] );
+
+			if ( ! Array.isArray( items ) ) {
+				break;
+			}
+
+			// eslint-disable-next-line guard-for-in
+			for ( const index in items ) {
+				const item = items[ index ];
+				const ecosystemTerm = item.ecosystem_types.pop();
+
+				if ( ecosystemTerm === pluginTerm ) {
+					// eslint-disable-next-line no-await-in-loop
+					const plugin = await this.preparePlugin( item );
+					if ( plugin ) {
+						this.plugins.push( plugin );
+					}
+				} else if ( ecosystemTerm === themeTerm ) {
+					// eslint-disable-next-line no-await-in-loop
+					const theme = await this.prepareTheme( item );
+					if ( theme ) {
+						this.themes.push( theme );
+					}
+				}
+			}
+
+			queryParams.page++;
+		} while ( queryParams.page <= totalPage );
+	}
+
+	/**
+	 * Store plugins and theme data in JSON respective file.
+	 *
+	 * @return {Promise<void>}
+	 */
+	async storeData() {
+		if ( this.plugins ) {
+			let output = await this.convertToPhpArray( this.plugins );
+			output = `<?php\nreturn ${ output };`;
+			fs.writeFileSync( 'includes/amp-plugins.php', output );
+		}
+
+		if ( this.themes ) {
+			let output = await this.convertToPhpArray( this.themes );
+			output = `<?php\nreturn ${ output };`;
+			fs.writeFileSync( 'includes/amp-themes.php', output );
+		}
+	}
+
+	/**
+	 * Execute given command in shell and return the output.
+	 *
+	 * @param {string} command Shell command.
+	 * @return {Promise<object>} Output or error from shell command.
+	 */
+	executeCommand( command ) {
+		return new Promise( ( done, failed ) => {
+			exec( command, ( error, stdout, stderr ) => {
+				if ( error ) {
+					error.stdout = stdout;
+					error.stderr = stderr;
+					failed( error );
+					return;
+				}
+				done( { stdout, stderr } );
+			} );
+		} );
+	}
+
+	/**
+	 * Convert JS object into PHP array variable.
+	 *
+	 * @param {Object} object An object that needs to convert into a PHP array.
+	 * @return {string|null} PHP array in string.
+	 */
+	async convertToPhpArray( object ) {
+		if ( 'object' !== typeof object ) {
+			return null;
+		}
+
+		const tempFilePath = '/tmp/amp.json';
+		const json = JSON.stringify( object );
+		const command = `php -r 'var_export( json_decode( file_get_contents( "${ tempFilePath }" ), true ) );'`;
+
+		fs.writeFileSync( tempFilePath, json );
+		const output = await this.executeCommand( command );
+
+		fs.unlinkSync( tempFilePath );
```

Comment:
Instead of writing out a temp file, can `stdin` be used?

```suggestion
		const command = `php -r 'var_export( json_decode( file_get_contents( "php://stdin" ), true ) );'`;
		const output = await this.executeCommand( command, JSON.stringify( object ) );
```

This would then require `executeCommand` to be updated to allow a second argument and when present to pass it into `exec` (how specifically I'm unsure of as I've not done so myself in Node).

Reply:
```js
let json = JSON.stringify( object );
json = json.replace( /'/gm, `\\'` );
const command = `ECHO '${ json }' | php -r 'var_export( json_decode( file_get_contents( "php://stdin" ), true ) );'`;
const output = await this.executeCommand( command );
```

I tried with the above snippet. However, It throws errors while echoing JSON data due to unexpected characters. ( I believe it's because the larger JSON string has a number of characters that aren't command friendly. )

Reply:
Right, you won't be able to `echo` it. You'd have to pass it to the `stdin` handle of the subprocess: https://stackoverflow.com/a/41343999/93579

Reply:
Thanks for the reference. 
I have changed the implementation. Turns out, `child_process.exec()` doesn't have option to pass `stdin` but `child_process.execSync()` have that option.